src/engine-actions/initiate-swap.spec.js

```diff
@@ -0,0 +1,95 @@
+const path = require('path')
+const { expect, rewire, sinon } = require('test/test-helper')
+
+const initiateSwap = rewire(path.resolve(__dirname, 'initiate-swap'))
+
+describe('initiateSwap', () => {
+  let networkAddressFormatter
+  let sendPayment
+  let swapHash
+  let address
+  let amount
+  let client
+  let engine
+  let maxTimeLock
+  let finalDelta
+
+  beforeEach(() => {
+    networkAddressFormatter = {
+      parse: sinon.stub().withArgs('fake address').returns({ publicKey: 'fake pubkey' })
+    }
+    sendPayment = sinon.stub().resolves({ paymentPreimage: 'fake preimage' })
+
+    initiateSwap.__set__('networkAddressFormatter', networkAddressFormatter)
+    initiateSwap.__set__('sendPayment', sendPayment)
+
+    swapHash = 'fake hash'
+    address = 'fake address'
+    amount = '100000'
+    maxTimeLock = 1201
+    finalDelta = 590
+    client = 'fake client'
+    engine = {
+      client,
+      logger: {
+        error: sinon.stub(),
+        info: sinon.stub()
+      },
+      secondsPerBlock: 600
+    }
+  })
+
+  it('sends a payment', async () => {
```

Comment:
Perhaps a matter of opinion, but I think this might be simpler if all the positive test cases were consolidated into a single test case.